### ISSUE TITLE
Add ability to provide your own HTTP Client

### ISSUE THREAD

**Issue by Hontaro** [josueeduardo/rest-client#85]
In Unirest you had to provide your own HTTPClient via `Unirest.setHttpClient(httpClient)`. This functionality has been lost with the RestClient.
The current ClientBuilder is missing customization options provided by the HttpClientBuilder such as setting your own SSL context.

**Comment by josueeduardo** [comment]
Yeah, it makes sense, the http-client API surface area is massive. The RestClient builder provides a very simplified configuration of the most used methods of it. I decided to stick with a minimal approach, and expand it on request basis. Will add SSL context for the builder and option to provide your own HttpClient on 1.6.1 release.
